### PR TITLE
Feature/tensorboard tf2

### PR DESCRIPTION
**Pull request by Davidnet**
Solves #47 and updates examples to show use hopefully this can remove the plt.show instances.

### CODE REVIEW

**Comment by Davidnet** [comment]
Checks fails due to jax and jaxlib missing @cgarciae I think you should specify that instalation in github action

File: examples/mnist_vae.py [hunk 1]
```diff
@@ -175,16 +182,23 @@ def main(debug: bool = False, eager: bool = False):
...
 
-    # plot results
-    plt.figure(figsize=(12, 12))
-    for i in range(5):
-        plt.subplot(2, 5, i + 1)
-        plt.imshow(x_sample[i], cmap="gray")
-        plt.subplot(2, 5, 5 + i + 1)
-        plt.imshow(y_pred["image"][i], cmap="gray")
+    # plot and save results
+    with SummaryWriter(logdir + "/val") as tbwriter:
```
Comment (charlielito) [inline examples/mnist_vae.py:186]: nice!

File: elegy/callbacks/tensorboard.py [hunk 2]
```diff
@@ -30,79 +27,109 @@ class TensorBoard(Callback):
...
+
         super(TensorBoard, self).__init__()
 
     def on_train_begin(self, logs=None):
-        self.writer = SummaryWriter(self.logdir, **self._open_args)
-
-    def on_train_batch_end(self, batch: int, logs):
-        if self.update_freq == "epoch":
+        self.train_writer = SummaryWriter(
+            self.logdir + "/train", purge_step=self.purge_step
```
Comment (charlielito) [inline elegy/callbacks/tensorboard.py:86]: Better to use `os.path.join`

File: elegy/callbacks/tensorboard.py [hunk 3]
```diff
@@ -30,79 +27,109 @@ class TensorBoard(Callback):
...
+        )
+        self.val_writer = SummaryWriter(
+            self.logdir + "/val", purge_step=self.purge_step
```
Comment (charlielito) [inline elegy/callbacks/tensorboard.py:89]: Better to use `os.path.join`

File: examples/mnist.py [hunk 4]
```diff
@@ -83,18 +90,24 @@ def call(self, image: jnp.ndarray):
...
-    plt.figure(figsize=(12, 12))
-    for i in range(3):
-        for j in range(3):
-            k = 3 * i + j
-            plt.subplot(3, 3, k + 1)
-
-            plt.title(f"{np.argmax(y_pred[k])}")
-            plt.imshow(x_sample[k], cmap="gray")
+    # plot and save results
+    with SummaryWriter(logdir + "/val") as tbwriter:
```
Comment (charlielito) [inline examples/mnist.py:94]: Better to use `os.path.join`

File: examples/mnist_autoencoder.py [hunk 5]
```diff
@@ -83,16 +89,25 @@ def call(self, x, y_pred):
...
 
-    # plot results
-    plt.figure(figsize=(12, 12))
-    for i in range(5):
-        plt.subplot(2, 5, i + 1)
-        plt.imshow(x_sample[i], cmap="gray")
-        plt.subplot(2, 5, 5 + i + 1)
-        plt.imshow(y_pred[i], cmap="gray")
+    # plot and save results
+    with SummaryWriter(logdir + "/val") as tbwriter:
```
Comment (charlielito) [inline examples/mnist_autoencoder.py:93]: Better to use `os.path.join`

File: examples/mnist_vae.py (hunk 1 above)
Comment (charlielito) [reply examples/mnist_vae.py:186]: Use `os.path.join`

File: examples/mnist_autoencoder.py (hunk 5 above)
Comment (cgarciae) [reply examples/mnist_autoencoder.py:93]: Lets use `pathlib.Path`

**Pull request merged**